### PR TITLE
Create testing functionality via temporary bots

### PR DESCRIPTION
**Pull request by essare-rimaz**
- before we implement changes, those should be tested
- some of these changes can only be evaluated visually within Telegram UI
- to give every tester the possibility to test these independently, they need a way how to create their own temporary chatbot
- since that is quite complicated, the creation/deletion of these temporary bots was automated via telethon library
- it still requires them to make changes with the credentials.yml file, however, it is still much easier than doing the entire process
- some changes were made specifically to counter the possibility of sharing sensitive information

### CODE REVIEW

**Review by kubasikus — COMMENTED** [review]
LGTM!

File: scripts/telethon_bot_creation.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# https://stackoverflow.com/questions/61432970/how-can-i-send-a-message-to-botfather-programmatically
+from telethon import TelegramClient, events
+from telethon_secrets import generate_bot_name, temporary_file
+from getpass import getpass
+
+
+api_id = getpass("API id:")
```
[the remark this answers is not in the document]
Comment (essare-rimaz) [reply scripts/telethon_bot_creation.py:10]: ✔️

File: scripts/telethon_secrets.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import names
+from datetime import datetime
+import os
+
+
+def generate_bot_name() -> str:
+    now = datetime.now()
+    bot_name = str(names.get_first_name())+"_"+str(now.year)+"_"+str(now.month)+"_"+str(now.day)+"_"+str(now.hour)+"_"+str(now.minute)
```
[the remark this answers is not in the document]
Comment (essare-rimaz) [reply scripts/telethon_secrets.py]: ✔️

File: scripts/telethon_bot_creation.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+async def message_handler(event):
+    if 'Please choose a name for your bot' in event.raw_text:
+        await event.reply(bot_name)
+    elif 'choose a username for your bot' in event.raw_text:
+        await event.reply(bot_user_name)
+    elif 'Done! Congratulations on your new bot' in event.raw_text:
+        print("Bot created!")
+        print(f"You can access it at {'https://t.me/'+bot_user_name}")
+        temporary_file(bot_name)
+        await client.disconnect()
```
[the remark this answers is not in the document]
Comment (essare-rimaz) [reply scripts/telethon_bot_creation.py:29]: ✔️

File: scripts/telethon_bot_deletion.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+
+@client.on(events.NewMessage)
+async def message_handler(event):
+    print(event.raw_text)
+    bot_name = read_bot_name()
+    print(bot_name)
+    if 'Choose a bot from the list below:' in event.raw_text:
+        await event.message.click(text=bot_name)
+        print(event.raw_text)
```
[the remark this answers is not in the document]
Comment (essare-rimaz) [reply scripts/telethon_bot_deletion.py:24]: ✔️

File: scripts/telethon_bot_deletion.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    if 'What do you want to do with the bot?' in event.raw_text:
+        await event.message.click(text='Delete Bot')
+    elif 'You are about to delete your bot' in event.raw_text:
+        await event.message.click(text='Yes, delete the bot')
+    elif 'Are you TOTALLY sure you want to delete' in event.raw_text:
+        await event.message.click(text="Yes, I'm 100% sure!")
+        print("bot was deleted")
+        # remove the name of the bot
+        remove_temp_dir()
+        await client.disconnect()
```
[the remark this answers is not in the document]
Comment (essare-rimaz) [reply scripts/telethon_bot_deletion.py:43]: ✔️

File: scripts/telethon_secrets.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    return bot_name
+
+
+def temporary_file(bot_name):
+    os.mkdir(os.path.join(os.getcwd(), "temp"))
```
[the remark this answers is not in the document]
Comment (essare-rimaz) [reply scripts/telethon_secrets.py]: ✔️

File: scripts/telethon_secrets.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+def read_bot_name():
+    with open("temp/botname.txt", "r") as f:
+        bot_name = f.readline()
+    return bot_name
+
+
+def remove_temp_dir():
+    os.remove(os.path.join(os.getcwd(), "temp", "botname.txt"))
+    os.rmdir(os.path.join(os.getcwd(), "temp"))
```
[the remark this answers is not in the document]
Comment (essare-rimaz) [reply scripts/telethon_secrets.py:26]: can be done via shutil, but i want to stick to os - keeping it as is

File: scripts/telethon_bot_creation.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
+# https://stackoverflow.com/questions/61432970/how-can-i-send-a-message-to-botfather-programmatically
```
[the remark this answers is not in the document]
Comment (essare-rimaz) [reply scripts/telethon_bot_creation.py:1]: ✔️

File: .gitignore [hunk 9]
```diff
@@ -3,3 +3,6 @@
 /.rasa
 /_class_learning
 /models
+/scripts/temp
+/scripts/session*
+credentials.yml
```
[the remark this answers is not in the document]
Comment (essare-rimaz) [reply .gitignore:16]: ✔️

**Pull request merged**